pkgs/tools/networking/norouter/default.nix

```diff
@@ -0,0 +1,31 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+  pname = "norouter";
+  version = "0.6.4";
+
+  src = fetchFromGitHub {
+    owner = "norouter";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0h5jzxm4fw50781zj76r5ksnxkzsnrygrykpa913v9nd24c09c7m";
+  };
+
+  vendorSha256 = "sha256-DZ2kcNV8AzNogAUTaeus4rz9gCFo0wm306jcz/cAj0M=";
+
+  subPackages = [ "cmd/norouter" ];
+  doInstallCheck = true;
+  installCheckPhase = ''
+    $out/bin/norouter --version | grep ${version} > /dev/null
```

Comment:
no tests?

Reply:
Well, there are upstream integration tests: https://github.com/norouter/norouter/tree/master/integration
But I'm not competent enough to implement them in nixos. I would tend to trust them upstream, though. Not sure if that's an acceptable stance.

What about unit-tests? (like: https://github.com/norouter/norouter/tree/master/cmd/norouter) Wouldn't they be already run by the nixpkgs instrumentation?

Reply:
nevermind, I overlooked the fact that this was just the install checks. That should be sufficient.